### ISSUE TITLE
[1.9.7] UMD incorrectly tree shaking module

### ISSUE THREAD

**Issue by Icestorm05** [parcel-bundler/parcel#1740]
# 🐛 bug report

When using the experimental scope hoisting argument on build, I'm getting the following error:
Uncaught TypeError: Object prototype may only be an Object or null: undefined
The error does not appear if I build without experimental scope hoisting.

## 🎛 Configuration (.babelrc, package.json, cli command)

Tsconfig.json:
```json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["./*"]
    },
    "outDir": "out-tsc",
    "module": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "strictNullChecks": true,
    "emitDecoratorMetadata": true,
    "target": "es6",
    "jsx": "react",
    "sourceMap": true
  }
}
```

## 🤔 Expected Behavior

It should build correctly.

## 😯 Current Behavior

This function was throwing the error:
```js
    var dq = Ba && Ba.__extends || function() {
        var e = Object.setPrototypeOf || {
            __proto__: []
        }instanceof Array && function(e, r) {
            e.__proto__ = r
        }
        || function(e, r) {
            for (var t in r)
                r.hasOwnProperty(t) && (e[t] = r[t])
        }
        ;
        return function(r, t) {
            function n() {
                this.constructor = r
            }
            e(r, t),
            r.prototype = null === t ? Object.create(t) : (n.prototype = t.prototype,
            new n)
        }
    }()
```

The function seems to be generated automatically by the typescript compiler, and is simulating single class inheritance.

```js
    var rm = function(e) {
        function r(r) {
            return e.call(this, sm(r).request) || this
        }
        return dq(r, e),
        r
    }(Qm);
```

Here, the function is called with an undefined value (e/Qm), hence the error. Following the stack trace, it brings me back to a file called httpLink.js in the apollo-link-http library.

httpLink.js before Parcel build:
```js
import { ApolloLink, Observable, fromError } from 'apollo-link';
return new ApolloLink(function (operation) {
...
}
```

In Parcel bundle:
```js
var lE, kE, jE, iE, hE, gE, pj;
...
return new pj(function(e) {
...
}
```

ApolloLink doesn't seem to appear in the bundle at all. There is just an undeclared variable in its place.

## 💁 Possible Solution

The problem seems to stem from the fact that importing just the ApolloProvider from react-apollo is tree shaking apollo-client out of the bundle.

```js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('react'), require('prop-types'), require('apollo-client')) :
    typeof define === 'function' && define.amd ? define(['exports', 'react', 'prop-types', 'apollo-client'], factory) :
    (factory((global['react-apollo'] = {}),global.React,global.PropTypes,global.apolloClient));
}(this, (function (exports,React,PropTypes,apolloClient) {
var ApolloProvider = ...
}
```

Since ApolloProvider does not use the apolloClient import here, it gets taken out. But, apollo-client is still required by the app:

```js
import { ApolloClient } from "apollo-client";
import { HttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { ApolloProvider } from "react-apollo";
```

The error seems to go away if I remove the apolloClient import in the UMD module, or if I import the ApolloProvider file directly.

## 🔦 Context

Couldn't build an application using Scope Hoisting and Apollo due to the problem.

## 💻 Code Sample

https://github.com/Icestorm05/react-apollo-test

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.7
| Node             | 10.6.0
| npm/Yarn         | 6.1.0
| Operating System | macOS 10.13.1 High Sierra

**Comment by sqs** [comment]
> @sqs Your issue could be the same as mine, could you check it out if you can?

Yes, mine is a different issue then. Thanks.

**Comment by devongovett** [comment]
Appears to be fixed by #2993.
